editor/src/components/canvas/canvas-strategies/strategies/grid-helpers.ts

```diff
@@ -604,3 +611,62 @@ function getGridPositionIndex(props: {
 }): number {
   return (props.row - 1) * props.gridTemplateColumns + props.column - 1
 }
+
+export function getGlobalFramesOfGridCellsFromMetadata(
+  metadata: ElementInstanceMetadata,
+): Array<Array<CanvasRectangle>> | null {
+  return getGlobalFramesOfGridCells(metadata.specialSizeMeasurements)
+}
+
+export function getGlobalFramesOfGridCells({
+  containerGridProperties,
+  rowGap,
+  columnGap,
+  padding,
+}: {
+  containerGridProperties: GridContainerProperties
+  rowGap: number | null
+  columnGap: number | null
+  padding: Sides
+}): Array<Array<CanvasRectangle>> | null {
+  const columnWidths = gridTemplateToNumbers(containerGridProperties.gridTemplateColumns)
+
+  const rowHeights = gridTemplateToNumbers(containerGridProperties.gridTemplateRows)
+
+  if (columnWidths == null || rowHeights == null) {
+    return null
+  }
+
+  const cellRects: Array<Array<CanvasRectangle>> = []
+  let yOffset = padding.top ?? 0
+  rowHeights.forEach((height) => {
+    let xOffset = padding.left ?? 0
+    const rowRects: CanvasRectangle[] = []
+    columnWidths.forEach((width) => {
+      const rect = canvasRectangle({ x: xOffset, y: yOffset, width: width, height: height })
+      rowRects.push(rect)
+      xOffset += width + (columnGap ?? 0)
```

Comment:
A small note - the gap shouldn't be added to the first offset of row/column

Reply:
It is not added to the first one, the offset is increased _after_ I added the first rect in line 647.